### ISSUE TITLE
When using the adapter, I encountered the issue "Module not found: Can't resolve 'aptos'."

### ISSUE THREAD

**Issue by caoyang2002** [aptos-labs/aptos-wallet-adapter#405]
# Link

https://aptos.dev/en/build/sdks/wallet-adapter/dapp

# Issues

When using the adapter, I encountered the following issue, and I hope to understand the reason why this problem was thrown.

```bash
./node_modules/@aptos-connect/wallet-adapter-plugin/dist/index.mjs:357:1
Module not found: Can't resolve 'aptos'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/@aptos-labs/wallet-adapter-react/node_modules/@aptos-labs/wallet-adapter-core/dist/index.mjs
./node_modules/@aptos-labs/wallet-adapter-react/dist/index.mjs
./src/app/layout.tsx
```

# code sandbox

[code sandbox
](https://codesandbox.io/p/devbox/github/codesandbox/sandbox-templates/tree/main/nextjs?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm096abms00063b7e8xuxpk7e%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm096abmr00023b7ewx3q0ev1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm096abms00043b7eq78olhgb%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm096abms00053b7e8sei0ohg%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm096abmr00023b7ewx3q0ev1%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm096abmr00013b7ej7f1go51%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cm096bpt400223b7euvsnhn8k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A23%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252Flayout.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cm096abmr00023b7ewx3q0ev1%2522%252C%2522activeTabId%2522%253A%2522cm096bpt400223b7euvsnhn8k%2522%257D%252C%2522cm096abms00053b7e8sei0ohg%2522%253A%257B%2522id%2522%253A%2522cm096abms00053b7e8sei0ohg%2522%252C%2522activeTabId%2522%253A%2522cm096aw0p00153b7ebs91f212%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522cm096aw0p00153b7ebs91f212%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm096abms00043b7eq78olhgb%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm096abms00033b7ebc9hq821%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522cm096abms00043b7eq78olhgb%2522%252C%2522activeTabId%2522%253A%2522cm096abms00033b7ebc9hq821%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

```bash
yarn dev
```

# Current status

- nextjs (app routor)

`package.json`

```json
{
  "name": "connect-wallet-adapter",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@aptos-labs/ts-sdk": "^1.27.1",
    "@aptos-labs/wallet-adapter-ant-design": "^3.0.11",
    "@aptos-labs/wallet-adapter-react": "^3.6.0",
    "next": "14.2.6",
    "petra-plugin-wallet-adapter": "^0.4.5",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.2.6",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
}
```

`src/app/layout.tsx`

```tsx
'use client'

import { PropsWithChildren } from 'react'
import { PetraWallet } from 'petra-plugin-wallet-adapter'
import { AptosWalletAdapterProvider } from '@aptos-labs/wallet-adapter-react'
import { WalletSelector } from '@aptos-labs/wallet-adapter-ant-design'
import React from 'react'
import {
  Aptos,
  AptosConfig,
  Network,
  APTOS_COIN,
  AccountAddressInput,
} from '@aptos-labs/ts-sdk'

const wallets = [new PetraWallet()]
export default function RootLayout({ children }: PropsWithChildren) {
  return (
    <body className="grow">
      <AptosWalletAdapterProvider
        plugins={wallets}
        autoConnect={true}
        dappConfig={{
          network: Network.TESTNET,
          aptosConnectDappId: 'dapp-id',
        }}
      >
        <WalletSelector />
        <div className="mt-16">{children}</div>
      </AptosWalletAdapterProvider>
    </body>
  )
}
```

**Comment by estellechoi** [comment]
Me too. the package is not compatible with Next.js 14

**Comment by yadsendeww** [comment]
Me too, can anyone fix this?

**Comment by o-az** [comment]
same issue here. You could try installing `aptos` manually first.
Oddly, `aptos` is a deprecated package: https://npm.im/aptos and this is supposed to be the replacement.

**Comment by caoyang2002** [comment]
It's possible that the error occurred because 'use client' was not used, as the rendering process for components with the adapter takes place entirely on the client side. If this marker is missing, an error will be reported.

https://github.com/Chyraw-Labs/aptos-case/tree/main/ts-case

**Comment by OlexBeatable** [comment]
> It's possible that the error occurred because 'use client' was not used, as the rendering process for components with the adapter takes place entirely on the client side. If this marker is missing, an error will be reported.
> 
> https://github.com/Chyraw-Labs/aptos-case/tree/main/ts-case

I don't think that 'use client' will fix it, because I have the same issue using React, which is fully rendered on client. The only thing which helped me - is manual installing aptos by adding it to the project dependencies.

**Comment by 0xmaayan** [comment]
This was due to old dependencies using the old and deprecated "aptos" SDK - should not be the case now
